### PR TITLE
feat: screenshot testing

### PR DESCRIPTION
**Pull request by niktverd**

### CODE REVIEW

**Review by NasgulNexus — APPROVED** [review]

File: playwright/playwright/index.tsx [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Import styles, initialize component theme here.
+// import '../src/common.css';
```
Comment (gorgeousvlad) [inline playwright/playwright/index.tsx:2]: remove line 2?

File: src/components/Anchor/__snapshots__/Anchor.visual.test.tsx-snapshots/Anchor-render-anchor-by-default-dark-chromium-linux.png
Comment (gorgeousvlad) [inline src/components/Anchor/__snapshots__/Anchor.visual.test.tsx-snapshots/Anchor-render-anchor-by-default-dark-chromium-linux.png:1]: I like Malevich, maybe we should add screenshot tests for components with visual presentation?)

**Review by gorgeousvlad — CHANGES_REQUESTED** [review]

File: playwright/playwright/index.tsx (hunk 1 above)
Comment (niktverd) [reply playwright/playwright/index.tsx:2]: ok

File: playwright/README.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
+# Playwright Test Component
+
+## How to write a test
+
+1. Select the component you want to write tests for
+2. Inside the component folder, create the `__tests__` folder and create a file inside it with the following name `<ComponentName>.visual.test.tsx`
+3. Writing a test:
+
+   > NOTE:
+   > Add `defaultDelay` between mounting and testing routine in every test. It is necessary because of animation in our package.
```
[the remark this answers is not in the document]
Comment (niktverd) [reply playwright/README.md:10]: ok

File: package.json [hunk 3]
```diff
@@ -123,6 +129,8 @@
     "@gravity-ui/stylelint-config": "^1.0.0",
     "@gravity-ui/tsconfig": "^1.0.0",
     "@gravity-ui/uikit": "^5.12.2",
+    "@playwright/experimental-ct-react": "^1.38.1",
```
[the remark this answers is not in the document]
Comment (niktverd) [reply package.json]: ok

File: playwright/README.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+   test('test description', async ({page, mount, expectScreenshot, defaultDelay}) => {
+     // mount the component
+     await mount(<MyComponent props={props} />);
+
+     // default delay to bypass animation issues
+     await defaultDelay();
+
+     // capture the screenshot
+     await expectScreenshot();
```
[the remark this answers is not in the document]
Comment (niktverd) [reply playwright/README.md:52]: the fixture inside itself makes all the arrangements to screenshot in different themes

File: src/components/Anchor/__snapshots__/Anchor.visual.test.tsx-snapshots/Anchor-render-anchor-by-default-dark-chromium-linux.png
Comment (niktverd) [reply src/components/Anchor/__snapshots__/Anchor.visual.test.tsx-snapshots/Anchor-render-anchor-by-default-dark-chromium-linux.png:1]: ok

File: playwright/playwright.config.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    workers: process.env.CI ? 8 : undefined,
+    /* Reporter to use. See https://playwright.dev/docs/test-reporters */
+    reporter,
+    /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
+    use: {
+        testIdAttribute: 'data-qa',
+        /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
+        trace: 'on-first-retry',
+        headless: true,
+        /* Port to use for Playwright component endpoint. */
```
Comment (Kyzyl-ool) [inline playwright/playwright.config.ts:52]: I think this comment is redundant

File: src/components/Button/__snapshots__/Button.visual.test.tsx-snapshots/Button-render-stories-Default-dark-chromium-linux.png
Comment (Kyzyl-ool) [inline src/components/Button/__snapshots__/Button.visual.test.tsx-snapshots/Button-render-stories-Default-dark-chromium-linux.png:1]: fyi: the image is broken

File: playwright/core/defaultDelay.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import {DEFAULT_MOUNT_TEST_DELAY} from './constants';
+import type {PlaywrightFixture} from './types';
+
+export const defaultDelayFixture: PlaywrightFixture<() => Promise<void>> = async ({page}, use) => {
+    const defaultDelay = async () => await page.waitForTimeout(DEFAULT_MOUNT_TEST_DELAY);
```
Comment (Kyzyl-ool) [inline playwright/core/defaultDelay.ts:5]: can we use smth like this?
```
await page.waitForLoadState('load');
```
waiting for `load` event can be more robust technique for a waiting for a page to load

**Review by gorgeousvlad — APPROVED** [review]

**Pull request merged**